southpark-search/tests/distributed/terraform.tf

```diff
@@ -0,0 +1,204 @@
+provider "aws" {
+  version = "~> 2.0"
+  region  = "us-east-2"
+}
+
+#Create repo
+resource "aws_ecr_repository" "southpark" {
+  name = "sp-repo"
+  tags = {
+    Name = "southpark_repo"
+  }
+}
+
+resource "aws_default_vpc" "default_vpc" {
+}
+
+resource "aws_instance" "encoder" {
+  ami           = "ami-07efac79022b86107"
+  instance_type = "t2.micro"
+  tags = {
+    Name = "encoder"
+  }
+}
+
+resource "aws_instance" "indexer" {
+  ami           = "ami-07efac79022b86107"
+  instance_type = "t2.micro"
+  tags = {
+    Name = "indexer"
+  }
+}
+
+resource "aws_instance" "flow" {
+  ami           = "ami-07efac79022b86107"
+  instance_type = "t2.micro"
+  tags = {
+    Name = "flow"
+  }
+}
+
+data "aws_subnet_ids" "default" {
+  vpc_id = "${aws_default_vpc.default_vpc.id}"
+}
+
+
+resource "aws_ecs_cluster" "southpark_cluster" {
+  name = "southpark_cluster"
+}
+
+resource "aws_iam_role_policy_attachment" "ecsTaskExecutionRole_policy" {
+  role       = "${aws_iam_role.ecsExecutionRole.name}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+resource "aws_iam_role" "ecsExecutionRole" {
+  name               = "ecsExecutionRole"
+  assume_role_policy = "${data.aws_iam_policy_document.assume_role_policy.json}"
+}
+
+data "aws_iam_policy_document" "assume_role_policy" {
+  statement {
+    effect = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+
+#Create task
+resource "aws_ecs_task_definition" "southpark_task" {
+  family                   = "southpark_task" 
+  container_definitions    = <<DEFINITION
+  [
+    {
+      "name": "southpark_task",
+      "image": "${aws_ecr_repository.southpark.repository_url}",
+      "essential": true,
+      "portMappings": [
+        {
+          "containerPort": 45678,
+          "hostPort": 45678
+        }
+      ],
+      "memory": 512,
```

Comment:
Last time I had to increase this because it was too little for the SP example, it works with this now?

Reply:
No problem in spinning up the instances with `512` memory, we can scale up if needed.

Reply:
The problem will only arise, once you run an actual flow on it. Let's see what happens